docs/model_developers/modules.rst

```diff
@@ -122,3 +122,60 @@ The format for requires in requirements.txt is the folllowing:
  * Or a repository location such as  git+https://github.com/project/python-foo The correct syntax
    to use is then: eggname@git+https://../repository#branch with branch being optional.
 
+
+Working on modules
+==================
+Since modules often have dependencies on other modules, it is common to develop against multiple
+modules (or a project and one or more modules) simultaneously. One might for example need to
+extend a dependency module to add support for some new feature. Because this use case is so common,
+this section will describe how to work on multiple modules simultaneously so that any changes are
+visible the compiler. This procedure is of course applicable for working on a single module as well.
+
+Setting up the dev environment
+------------------------------
+v1 modules
+^^^^^^^^^^
+A v1 module is a module that is not packaged and published to a pip index but is solely distributed
+via git. V1 modules are installed on the fly by the compiler and can be found in the project's
+`libs` directory. Any changes you make to the module source in the `libs` dir will be reflected in
+the next compile. If you need to be able to make changes to multiple modules simultaneously, the
+simplest way to set up your development environment is to create a project that imports the desired
+modules, then run a compile against it. Alternatively, you can manually clone the module repos.
+
+v2 modules
+^^^^^^^^^^
+A v2 module in development form has mostly the same structure as a v1 module. The main difference is
+that it is meant to be published as a Python package. A project or another v2 module then lists all
+required v2 modules as dependencies in its `pyproject.toml`. The compiler does not install v2
+modules on the fly. In line with how Python depdencies work in general they are expected to be
+installed in advance. As a result they will not be placed in the project's `libs` directory.
+
+To set up your development environment, first clone all modules you wish to develop against and
+install them in editable mode with `poetry install`. This will fetch all its module v2 dependencies
+from the Python package index and install them in the active Python environment. Any modules you
```

Comment:
Shouldn't we mention that the user should update the `pyproject.toml` file in the following way when a editable install is required on a project:

```
[tool.poetry.dependencies]
my-package = {path = "../my/path", develop = true}
```

So point to the source directory instead of a python package and set the develop flag to true.

Reply:
Since we're moving away from poetry I assume I can disregard this comment?

Reply:
yes